pangeo_forge_recipes/recipes/xarray_zarr.py

```diff
@@ -160,11 +168,41 @@ def cache_input_metadata(
         xarray_open_kwargs=xarray_open_kwargs,
         delete_input_encoding=delete_input_encoding,
         process_input=process_input,
+        open_input_with_fsspec_reference=open_input_with_fsspec_reference,
+        metadata_cache=metadata_cache,
     ) as ds:
         input_metadata = ds.to_dict(data=False)
         metadata_cache[_input_metadata_fname(input_key)] = input_metadata
 
 
+def make_input_reference(
+    input_key: InputKey,
+    metadata_cache: MetadataTarget,
+    file_pattern: FilePattern,
+    input_cache: Optional[CacheFSSpecTarget],
+    copy_input_to_local_file: bool,
+) -> None:
+
+    fname = file_pattern[input_key]
+    if input_cache is None:
+        protocol = fsspec.utils.get_protocol(fname)
+        url = unstrip_protocol(fname, protocol)
+    else:
+        url = unstrip_protocol(input_cache._full_path(fname), input_cache.fs.protocol)
+    with file_opener(
+        fname,
+        cache=input_cache,
+        copy_to_local=copy_input_to_local_file,
+        bypass_open=False,
+        secrets=file_pattern.query_string_secrets,
+        **file_pattern.fsspec_open_kwargs,
+    ) as fp:
+        ref_data = create_hdf5_reference(fp, url, fname)
+
+    ref_fname = _input_reference_fname(input_key)
+    metadata_cache[ref_fname] = ref_data
```

Comment:
Maybe this is a recipe detail, but it would be good to figure out where we can store these reference files for the future, perhaps together with a hash/uid of the file they were created from.